zebra-chain/src/types/amount.rs

```diff
@@ -32,6 +128,17 @@ where
     }
 }
 
+impl<C> TryFrom<i32> for Amount<C>
+where
+    C: AmountConstraint,
+{
+    type Error = Error;
```

Comment:
This impl requires a `Result` with an `Error` type. Together with the orphan impl problem we ran into, I think that this suggests that the `ops` impls should be `Result`-based rather than `Option`-based, for two reasons:

1. It allows writing, e.g., `a + b.try_into()` where `a: Amount`, `b: i32`;

2. It allows writing `(a + b)?` to cleanly handle failures.

Reply:
Otherwise we're stuck in a situation where some parts use Options and other parts use Results.

Reply:
big agree